### DISCUSSION TITLE
python wrapper

### DISCUSSION THREAD

**Discussion by Viste** [azerothcore/azerothcore-wotlk#5378] [Ideas]
python scripts wrapper for core? Python in server cli?

**Comment by FrancescoBorzi** [comment]
what benefit would that bring? our current cli is written in bash even though we'd like to slowly migrate it to [deno](https://deno.land/)

**Reply by Viste** [reply]
In case of scripts, python just more easy then C++ and more powerfull then lua )
in case of server cli, python can make it very powerful, for example you can write some small script to get some vars from db direct in erver cli and then do something else

**Reply by alchem1ster** [reply]
Python is almost twice as slow as Lua. Also, it is heavier than Lua and even more so Bash. 
Yes, it is flexible, but I personally don't see any advantages of such a move.
If you want power, simplicity and speed, then you could look at Nim or even Go.
_p.s. Senior Python dev._

**Reply by bloodrose00** [reply]
why did you choose deno？easy to deploy？by the way，Docker setup is very cool！

**Reply by Viste** [reply]
> Python is almost twice as slow as Lua. 

yes but also more powerful

> If you want power, simplicity and speed, then you could look at Nim or even Go.

GO eat to more RAM and need garbage collector and etc, Nim first time when i hear about this :)
